### PR TITLE
Working Validation

### PR DESCRIPTION
**Pull request by srujanr40**
- Prevents the Create Session form from being submitted unless all required fields are filled out, and date/time is set for a time beyond current time
- Blue "+ Create Session" button from the Dashboard now disappears after clicked, and reappears after form is submitted (avoids confusion as the button still looked like it should be functional)
Screenshot 2023-07-02 at 4 52 38 PM
Screenshot 2023-07-02 at 4 53 43 PM

### CODE REVIEW

File: client/src/client/components/CreateSession/CreateSessionTextFields.jsx [hunk 1]
```diff
@@ -83,6 +86,24 @@ export default function CreateSessionTextFields(props) {
     props.closeModal();
   };
 
+  function validityCheck() {
+    console.log(session_date_time.unix() + "    " + Date.now());
```
Comment (rkelly123) [inline client/src/client/components/CreateSession/CreateSessionTextFields.jsx]: This here on purpose?

File: client/src/client/components/Dashboard/Dashboard.jsx [hunk 2]
```diff
@@ -1,92 +1,114 @@
...
-    const [isSessionMoreInfoModalOpen, setIsSessionMoreInfoModalOpen] = useState(false);
-    const [selectedSessionId, setSelectedItemId] = useState(null);
-
-    const sessions = useSelector(store => store.sessionReducer).sessions;
+  const dispatch = useDispatch();
 
+  const [isCreateSessionModalOpen, setIsCreateSessionModalOpen] =
+    useState(false);
+  const [isSessionMoreInfoModalOpen, setIsSessionMoreInfoModalOpen] =
+    useState(false);
```
Comment (rkelly123) [inline client/src/client/components/Dashboard/Dashboard.jsx:22]: This on purpose or is it from a linter? Just wondering why we're putting the useState on a new line?

File: server/routes/session.js [hunk 3]
```diff
@@ -157,7 +157,7 @@ const featuredSessions = [
 
 // GET all sessions
 router.get('/', function (req, res, next) {
-    if (req.query) {
+    if (req.query.sport) {
         let filteredSessions = sessions.filter(session => session.sport === req.query.sport)
         return res.status(200).send(filteredSessions)
     }
```
Comment (rkelly123) [inline server/routes/session.js:163]: How did this bug slip through lol.

File: client/src/client/components/Dashboard/Dashboard.jsx (hunk 2 above)
Comment (srujanr40) [reply client/src/client/components/Dashboard/Dashboard.jsx:22]: `Prettier`'s code formatting. I think the line was too long

File: server/routes/session.js (hunk 3 above)
Comment (srujanr40) [reply server/routes/session.js:163]: 😭 🙇🏼

File: client/src/client/components/CreateSession/CreateSessionTextFields.jsx (hunk 1 above)
Comment (srujanr40) [reply client/src/client/components/CreateSession/CreateSessionTextFields.jsx]: Nope. Removed

**Review by edvinkeskin — APPROVED** [review]
I did a fresh start and ran, web app works, no visible errors via small test run

**Pull request merged**